tool/tsh/db.go

```diff
@@ -834,59 +792,44 @@ func onDatabaseConnect(cf *CLIConf) error {
 	return nil
 }
 
-// getDatabaseInfo fetches information about the database from tsh profile is DB is active in profile. Otherwise,
-// the ListDatabases endpoint is called.
-func getDatabaseInfo(cf *CLIConf, tc *client.TeleportClient) (*tlsca.RouteToDatabase, types.Database, error) {
-	database, err := pickActiveDatabase(cf)
-	if err == nil {
-		switch database.Protocol {
-		case defaults.ProtocolCassandra:
-			// Cassandra CLI connection require database resource to determine
-			// if the target database is AWS hosted in order to skip the password prompt.
```

Comment:
(I was about to ask a question but halfway through I realized that the code is correct, I'm leaving the comment anyway for any future readers)

This logic used to be executed whenever `getDatabaseInfo` was called, that is in `onDatabaseConnect` and `onProxyCommandDB`. Now it has been moved to `maybeAddDBUserPassword` which is called from both of those places.

Reply:
yeah so before this change we had an indirect dependency on this `getDatabaseInfo` behavior in the `maybeAddDBUserPassword` function, since we needed to have the `types.Database` fetched even if there is an active profile. Now `maybeAddDBUserPassword` just gets the database if it wasn't already loaded before, so it's the same behavior